client/my-sites/plans-features-main/components/test/plan-notice.tsx

```diff
@@ -0,0 +1,171 @@
+		mUsePlanUpgradeCreditsDisplay.mockImplementation( () => ( {
+			creditsValue: 100,
+			isPlanUpgradeCreditEligible: true,
+		} ) );
+	} );
+
+	test( 'A contact site owner <PlanNotice /> should be shown no matter what other conditions are met, when the current site owner is not logged in, and the site is on the plan is paid', () => {
+		mGetDiscountByName.mockImplementation( () => discount );
+		mUsePlanUpgradeCreditsDisplay.mockImplementation( () => ( {
+			creditsValue: 100,
+			isPlanUpgradeCreditEligible: true,
+		} ) );
+		mIsCurrentPlanPaid.mockImplementation( () => true );
+		mIsCurrentUserCurrentPlanOwner.mockImplementation( () => false );
+
+		renderWithProvider(
+			<PlanNotice
+				discountInformation={ { withDiscount: 'test', discountEndDate: new Date() } }
+				visiblePlanNames={ plansList }
+				isInSignup={ false }
+				siteId={ 10000000 }
+			/>
+		);
+		expect( screen.getByRole( 'status' ).textContent ).toBe(
+			'This plan was purchased by a different WordPress.com account. To manage this plan, log in to that account or contact the account owner.'
+		);
+	} );
+
+	test( 'A discount <PlanNotice /> should be shown if the user is the site owner and no matter what other conditions are met', () => {
+		mIsCurrentUserCurrentPlanOwner.mockImplementation( () => true );
+		mIsCurrentPlanPaid.mockImplementation( () => true );
+		mGetDiscountByName.mockImplementation( () => discount );
+		mUsePlanUpgradeCreditsDisplay.mockImplementation( () => ( {
+			creditsValue: 100,
+			isPlanUpgradeCreditEligible: true,
+		} ) );
+
+		renderWithProvider(
+			<PlanNotice
+				discountInformation={ { withDiscount: 'test', discountEndDate: new Date() } }
+				visiblePlanNames={ plansList }
+				isInSignup={ false }
+				siteId={ 32234 }
+			/>
+		);
+		expect( screen.getByRole( 'status' ).textContent ).toBe( discount.plansPageNoticeText );
+	} );
+
+	test( 'A plan upgrade credit <PlanNotice /> should be shown in a site where a plan is purchased, without other active discounts, has upgradeable plan and, the site owner is logged in', () => {
+		mIsCurrentUserCurrentPlanOwner.mockImplementation( () => true );
+		mIsCurrentPlanPaid.mockImplementation( () => true );
+		mGetDiscountByName.mockImplementation( () => false );
+		mUsePlanUpgradeCreditsDisplay.mockImplementation( () => ( {
+			creditsValue: 100,
+			isPlanUpgradeCreditEligible: true,
+		} ) );
+
+		renderWithProvider(
+			<PlanNotice
+				discountInformation={ { withDiscount: 'test', discountEndDate: new Date() } }
+				visiblePlanNames={ plansList }
+				isInSignup={ false }
+				siteId={ 32234 }
+			/>
+		);
+		expect( screen.getByRole( 'status' ).textContent ).toBe(
+			'You have $100.00 of pro-rated credits available from your current plan. Apply those credits towards an upgrade before they expire!'
+		);
+	} );
+
+	test( 'A marketing message <PlanNotice /> when no other notices are available and marketing messages are available', () => {
```

Comment:
Obviously, we expect `MarketingMessage` to _not_ take precedence judging from this test (yet it always does for me in the app) 🤔

Reply:
I actually checked this and the UI seem to honour the precedence 🤔 
![notices](https://user-images.githubusercontent.com/3422709/235243281-088ab04a-6c09-4437-ae26-2af1b2f831a8.gif)

Reply:
Cool. Thanks for confirming! was probably doing something wrong on my end then. Sorry 🤷